os/arch/arm/src/tiva/lm3s_ethernet.c

```diff
@@ -797,8 +688,19 @@ static void tiva_receive(struct tiva_driver_s *priv)
 			 */
 
 			EMAC_STAT(priv, rx_ip);
-			arp_ipin(&priv->ld_dev);
-			ipv4_input(&priv->ld_dev);
+
+		while (1) {
+			int res = ethernetif_input(&priv->ld_dev);
+			if (res == 0) {
+				break;
+			}
+			else{
+				printf("trying to handle again\n");
+			}
+			usleep(2000);
```

Comment:
There is an overhead from printf above, followed by a 2 ms sleep. Please add as comments, the reasoning behind this statement, would help future maintainence.